src/ray/core_worker/experimental_mutable_object_provider.cc

```diff
@@ -74,20 +84,14 @@ void MutableObjectProvider::HandleRegisterMutableObject(
     const ObjectID &reader_object_id) {
   absl::MutexLock guard(&remote_writer_object_to_local_reader_lock_);
 
-  if (remote_writer_object_to_local_reader_.count(writer_object_id)) {
-    // Channel already exists.
-    remote_writer_object_to_local_reader_[writer_object_id].num_readers += num_readers;
-  } else {
-    // Channel does not exist.
-    LocalReaderInfo info;
-    info.num_readers = num_readers;
-    info.local_object_id = reader_object_id;
-    bool success =
-        remote_writer_object_to_local_reader_.insert({writer_object_id, info}).second;
-    RAY_CHECK(success);
-
-    RegisterReaderChannel(reader_object_id);
-  }
+  LocalReaderInfo info;
```

Comment:
Why the change from the previous code here?

Reply:
Happy to change it back, but it seemed cleaner to make this idempotent rather than change the functionality based on whether the mutable object has already been scheduled.

Reply:
No this is alright; that was just me skimming quickly and not being to immediately tell what the difference was.